ckanext/stadtzhharvest/tests/test_harvester.py

```diff
@@ -50,3 +50,313 @@ def test_load_metadata_from_path(self):
+
+
+class TestStadtzhHarvestFunctional(FunctionalHarvestTest):
+
+    def test_harvest_create_test_dropzone(self):
+        data_path = os.path.join(
+            __location__,
+            'fixtures',
+            'test_dropzone'
+        )
+        test_config = json.dumps({
+            'data_path': data_path,
+            'metafile_dir': '',
+            'metadata_dir': 'test-metadata',
+            'update_datasets': True,
+            'update_date_last_modified': False
+        })
+
+        results = self._test_harvest_create(1, config=test_config)['results']
+        eq_(len(results), 1)
+        eq_(results[0]['title'], u'Administrative Einteilungen Stadt Zürich')
+
+    def test_harvest_create_dwh(self):
+        data_path = os.path.join(
+            __location__,
+            'fixtures',
+            'DWH'
+        )
+        test_config = json.dumps({
+            'data_path': data_path,
+            'metafile_dir': '',
+            'metadata_dir': 'dwh-metadata',
+            'update_datasets': True,
+            'update_date_last_modified': False
+        })
+
+        results = self._test_harvest_create(3, config=test_config)
+        eq_(len(results['results']), 3)
+        for result in results['results']:
+            expected_titles = [
+                u'Geburten nach Jahr, Geschlecht und Stadtquartier',
+                u'Test Nachnamen in der Stadt Zürich',
+                u'Daten der permanenten Velozählstellen - Stundenwerte',
+            ]
+
+            assert result['title'] in expected_titles, "Title does not match result: %s" % result
+
+    def test_harvest_create_geo(self):
+        data_path = os.path.join(
+            __location__,
+            'fixtures',
+            'GEO'
+        )
+        test_config = json.dumps({
+            'data_path': data_path,
+            'metafile_dir': 'DEFAULT',
+            'metadata_dir': 'geo-metadata',
+            'update_datasets': False,
+            'update_date_last_modified': True
+        })
+
+        results = self._test_harvest_create(2, config=test_config)
+        eq_(len(results['results']), 2)
+        for result in results['results']:
+            expected_titles = ['Alterswohnung', 'Amtshaus']
+            assert result['title'] in expected_titles, "Title does not match result: %s" % result
+
+    def _test_harvest_create(self, num_objects, **kwargs):
+        harvest_source = self._create_harvest_source(**kwargs)
+
+        self._run_full_job(harvest_source['id'], num_objects=num_objects)
+
+        # Check that two datasets were created
+        fq = "+type:dataset harvest_source_id:{0}".format(harvest_source['id'])
+        results = h.call_action('package_search', {}, fq=fq)
+        eq_(results['count'], num_objects)
+        return results
+
+    def test_harvest_update_dwh(self):
+        data_path = os.path.join(
+            __location__,
+            'fixtures',
+            'DWH'
+        )
+        temp_dir = tempfile.mkdtemp()
```

Comment:
In the setup()-method of FunctionalHarvestTest there is already a temp-dir created. No need to create it twice.

Reply:
done.